libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -3070,6 +3070,13 @@ MAV_RESULT GCS_MAVLINK::set_message_interval(uint32_t msg_id, int32_t interval_u
     } else {
         interval_ms = interval_us / 1000;
     }
+#if AP_SCHEDULER_ENABLED
+    // message rate can't be greate than 0.8 * SCHED_LOOP_RATE
+    if (interval_ms*800 < AP::scheduler().get_loop_period_us()) {
```

Comment:
Might be nice to have a constant or a function here so it doesn't get out of sync with the logic in `set_ap_message_interval`.

Reply:
Sure. I've now pulled it out to ``libraries/GCS_MAVLink/GCS.h``